.github/workflows/build.yml

```diff
@@ -5,45 +5,77 @@ env:
 jobs:
   build:
     runs-on: ubuntu-22.04
+    permissions:
+      packages: write
+      contents: read
     steps:
       - name: Checkout Source Code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v2
 
       - name: Set up Docker Buildx
         id: buildx
-        uses: docker/setup-buildx-action@v1
+        uses: docker/setup-buildx-action@v2
 
       - name: Docker meta
         id: docker_meta
-        uses: docker/metadata-action@v3
+        uses: docker/metadata-action@v4
         with:
-          images: ${{ secrets.DOCKER_ORGANIZATION }}/toolchain
+          images: |
+            docker.io/${{ github.repository_owner }}/toolchain
+            ghcr.io/${{ github.repository_owner }}/toolchain
           tags: |
             type=ref,event=branch
             type=semver,pattern={{version}}
 
       - name: Login to Docker Hub
-        uses: docker/login-action@v1
+        uses: docker/login-action@v2
+        if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/release/') || startsWith(github.ref, 'refs/tags/v')
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
 
+      - name: Login to GitHub Container Registry
+        if: github.ref != 'refs/heads/main' && !startsWith(github.ref, 'refs/heads/release/') && !startsWith(github.ref, 'refs/tags/v')
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
       - name: Download linux
         run: make download
 
-      - name: Build and push docker image
-        id: docker_build
-        uses: docker/build-push-action@v2
+      - name: Set up Depot CLI
+        uses: depot/setup-action@v1
+
+      - name: Build and push docker image to GitHub Container Registry
```

Comment:
You don't need a separate step to publish to GHCR.
It's better to control in the meta step using `enable`

Reply:
It tried to publish on docker hub because the meta is wrong